### PR TITLE
Address fixes

### PR DESCRIPTION
**Pull request by alejandro-isaza**

### CODE REVIEW

File: tests/AddressTests.cpp [hunk 1]
```diff
@@ -31,4 +31,9 @@ TEST(Address, ValidateLitecoin) {
     EXPECT_FALSE(validate("1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2", TWCoinTypeLitecoin));
 }
 
+TEST(Address, ValidateBitcoinCash) {
+    EXPECT_TRUE(validate("bitcoincash:qruxj7zq6yzpdx8dld0e9hfvt7u47zrw9gfr5hy0vh", TWCoinTypeBitcoinCash));
+    EXPECT_FALSE(validate("1BvBMSEYstWetqTFn5Au4m4GFg7xJaNVN2", TWCoinTypeBitcoinCash));
```
Comment (vikmeup) [inline tests/AddressTests.cpp]: @hewigovens what was the case for bitcoin cash where user wanted to send to `1` address?

File: src/Address.cpp [hunk 2]
```diff
@@ -113,10 +115,10 @@ bool TW::validate(const std::string& string, TWCoinType coin) {
         return Bitcoin::Address::isValid(string, {0x41});
 
     case TWCoinTypeZcoin:
-        return Bitcoin::Address::isValid(string, {0x52, 0x07});
+        return Bitcoin::Address::isValid(string, {TWP2PKHPrefixZcoin, TWP2SHPrefixZcoin});
```
Comment (vikmeup) [inline src/Address.cpp]: 👍

File: tests/AddressTests.cpp (hunk 1 above)
Comment (hewigovens) [reply tests/AddressTests.cpp]: This line should be true

File: src/Address.cpp [hunk 3]
```diff
@@ -72,10 +74,10 @@ std::string TW::loadAddress(const Data& data, TWCoinType coin) {
...
     case TWCoinTypeBinance:
-        return Tendermint::Address::isValid(string, HRP_BINANCE);
+        return Tendermint::Address::isValid(string, HRP_BINANCE) || Tendermint::Address::isValid(string, HRP_BINANCE_TEST);
 
     case TWCoinTypeBitcoin:
-        return Bitcoin::Bech32Address::isValid(string, HRP_BITCOIN) || Bitcoin::Address::isValid(string, {0x00, 0x05});
+        return Bitcoin::Bech32Address::isValid(string, HRP_BITCOIN) || Bitcoin::Address::isValid(string, {TWP2PKHPrefixBitcoin, TWP2SHPrefixBitcoin});
 
     case TWCoinTypeBitcoinCash:
         return Bitcoin::CashAddress::isValid(string);
```
Comment (hewigovens) [inline src/Address.cpp]: `Bitcoin::CashAddress::isValid(string) || Bitcoin::Address::isValid`

Comment (alejandro-isaza) [reply src/Address.cpp]: what prefix?

**Pull request merged**